core/templates/base-components/oppia-footer.component.ts

```diff
@@ -53,6 +63,40 @@ export class OppiaFooterComponent {
       return 'https://medium.com/oppia-org';
     }
   }
+
+  validateEmailAddress(): boolean {
+    let regex = new RegExp(AppConstants.EMAIL_REGEX);
+    return regex.test(String(this.emailAddress));
+  }
+
+  subscribeToMailingList(): void {
+    this.mailingListBackendApiService.subscribeUserToMailingList(
+      String(this.emailAddress),
+      // Most of the times, since the name is not provided
+      // so it is null. In such case placeholder name is set to UserX.
+      String(this.name) ?? 'UserX',
+      AppConstants.MAILING_LIST_WEB_TAG
+    ).then((status) => {
+      if (status) {
+        this.alertsService.addInfoMessage('Done!', 1000);
```

Comment:
I'm not entirely sure if this will lead to the right UI. Could you please show videos with each of these info messages showing so we can see the user experience?

Reply:
This seems to work properly with proper UI. I am attaching a video of it with different info messages. Please have a look

https://github.com/oppia/oppia/assets/76906722/878e0ed6-d9eb-4d75-85e1-191b4c4124b5

PS: Not a part of this PR though, but the `thanks-for-subscribing-modal` does not seem right in the dimension `768-991`.

Reply:
This looks good, thanks. Could you file a separate issue for the modal problem please?

Reply:
This looks good, thanks. Could you file a separate issue for the modal problem please?

Reply:
Done, opened an issue for it. also, the same issue seemed to happen in the `donating-modal`. #18709